src/coreclr/jit/gentree.cpp

```diff
@@ -23176,7 +23176,7 @@ GenTree* Compiler::gtNewSimdShuffleNode(
 #if defined(TARGET_XARCH)
     uint8_t  control   = 0;
     bool     crossLane = false;
-    bool     needsZero = varTypeIsSmallInt(simdBaseType);
+    bool     needsZero = varTypeIsSmallInt(simdBaseType) && (simdSize != 64);
```

Comment:
This is correct for the PR, but just noting we'll want to update the other `simdSizes` to use the full-width permutations as well, long term.